File: Low Level Design (LLD)/DesignPatterns/3. Structural/Strategy/Java/PaymentService.java
public class PaymentService {
    private PaymentStrategy strategy;

    public void setPaymentStrategy(PaymentStrategy strategy) {
        this.strategy = strategy;
    }

    public void processPayment(int amount) {
        if (strategy == null) {
            System.out.println("Please set a payment strategy.");
            return;
        }
        strategy.pay(amount);
    }
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Strategy/Cpp/src/PaymentService.cpp
#include "../include/PaymentService.h"
#include <iostream>

PaymentService::PaymentService() : strategy(nullptr) {}

void PaymentService::setPaymentStrategy(PaymentStrategy* strategy) {
    this->strategy = strategy;
}

void PaymentService::processPayment(int amount) {
    if (strategy)
        strategy->pay(amount);
    else
        std::cout << "No payment strategy set." << std::endl;
}
